src/Operation/Update.php

```diff
@@ -167,40 +167,32 @@ public function execute(Server $server)
             throw UnsupportedException::collationNotSupported();
         }
 
-        $updateOptions = [
-            'multi' => $this->options['multi'],
-            'upsert' => $this->options['upsert'],
-        ];
-
-        if (isset($this->options['arrayFilters'])) {
-            $updateOptions['arrayFilters'] = $this->options['arrayFilters'];
-        }
-
-        if (isset($this->options['collation'])) {
-            $updateOptions['collation'] = (object) $this->options['collation'];
-        }
-
         $bulkOptions = [];
 
         if (isset($this->options['bypassDocumentValidation']) && \MongoDB\server_supports_feature($server, self::$wireVersionForDocumentLevelValidation)) {
             $bulkOptions['bypassDocumentValidation'] = $this->options['bypassDocumentValidation'];
         }
 
         $bulk = new Bulk($bulkOptions);
-        $bulk->update($this->filter, $this->update, $updateOptions);
+        $bulk->update($this->filter, $this->update, $this->createUpdateOptions());
 
-        $writeResult = $server->executeBulkWrite($this->databaseName . '.' . $this->collectionName, $bulk, $this->createOptions());
+        $writeResult = $server->executeBulkWrite($this->databaseName . '.' . $this->collectionName, $bulk, $this->createExecuteOptions());
 
         return new UpdateResult($writeResult);
     }
 
+    public function getCommandDocument()
+    {
+        return ['update' => $this->collectionName, 'updates' => [['q' => $this->filter] + ['u' => $this->update] + $this->createUpdateOptions()]];
```

Comment:
`['q' => $this->filter] + ['u' => $this->update]` can be reduced to a single array.